test/hotspot/jtreg/compiler/cha/AbstractRootMethod.java

```diff
@@ -50,12 +50,23 @@
  */
 package compiler.cha;
 
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+
 import static compiler.cha.Utils.*;
 
 public class AbstractRootMethod {
     public static void main(String[] args) {
         run(AbstractClass.class);
         run(AbstractInterface.class);
+
+        // Implementation limitation: CHA is not performed by C1 during inlining through MH linkers.
+        if (!sun.hotspot.code.Compiler.isC1Enabled()) {
```

Comment:
I think you also need to check that C2 is enabled (could be JVMCI).

Reply:
Until JVMCI/Graal is migrated to vtable-based CHA, the tests are run only with C1/C2:

```
@requires !vm.graal.enabled & vm.opt.final.UseVtableBasedCHA == true
```